editor/gui/editor_run_bar.cpp

```diff
@@ -356,14 +360,39 @@ HBoxContainer *EditorRunBar::get_buttons_container() {
 	return main_hbox;
 }
 
+void EditorRunBar::_instance_starting(int p_idx, List<String> &r_arguments) {
+	singleton->_instance_starting_internal(p_idx, r_arguments);
+}
+
+void EditorRunBar::_instance_starting_internal(int p_index, List<String> &r_arguments) {
+	Array arguments;
+
+	// We need to convert the an Array so it can be passed as parameter in a signal.
+	for (const String &arg : r_arguments) {
+		arguments.push_back(arg);
+	}
+
+	emit_signal(SNAME("instance_starting"), p_index, arguments);
```

Comment:
Wouldn't it be better to set the callback directly from GameView instead of indirecting via signal? It's not used anywhere else and you could avoid conversion to Array.

Reply:
I'm not sure I understand what you mean, can you elaborate a bit please? My concern was to prevent referencing a plugin from `editor_run_bar`.

Reply:
You can set `instance_starting_callback` to a method from GameView, so it can be called directly.

Reply:
Done!